samza-core/src/main/java/org/apache/samza/zk/ProcessorData.java

```diff
@@ -51,4 +54,17 @@ public String getHost() {
   public String getProcessorId() {
     return processorId;
   }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(processorId, host);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null) return false;
+    if (getClass() != obj.getClass()) return false;
+    final ProcessorData other = (ProcessorData) obj;
+    return processorId.equals(other.processorId) && host.equals(other.host);
```

Comment:
I don't think we should be comparing host for equality. It should only be processors. For example, a case where the duplicate processorId comes on a different host, but it is still a duplicate as far as partitioning/grouping is concerned. Please correct me if I am wrong.